src/models/user.ts

```diff
@@ -0,0 +1,142 @@
+import { Sequelize, Model, DataTypes, NOW, Op } from "sequelize";
+
+const sequelize = new Sequelize(
+  `postgres://${process.env.DB_USER}:${process.env.DB_PASS}@${process.env.DB_IP}/${process.env.DB_NAME}`
+);
+
+export class User extends Model {
+  public id!: number;
+  public username!: string;
+  public password!: string;
+  public email!: string;
+  public userRank!: string;
+  public userAvatar!: string;
+  public createdAt!: Date;
+  public updatedAt!: Date;
+  public authenticated!: boolean;
+  public authCodeId!: number;
+}
+
+User.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    userRank: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "Studenciak",
+    },
+    userAvatar: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue:
+        "https://res.cloudinary.com/survikrowa/image/upload/v1597051975/sq6iqojgm6qlm5lqcifv.jpg",
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: NOW,
+    },
+    authenticated: {
+      type: DataTypes.BOOLEAN,
+      defaultValue: false,
+    },
+    authCodeId: {
+      type: DataTypes.NUMBER,
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "mzg_backend",
+    sequelize,
+  }
+);
+
+export const createUser = async (
+  { username, password, email }: Partial<User>,
+  authCodeId: number
+) => {
+  try {
+    await User.create({
+      username,
+      password,
+      email,
+      authCodeId,
+    });
+  } catch (error) {
+    console.error(error);
```

Comment:
O jeszcze lepiej log :D A tak na poważnie to nie jest zły pomysł tylko dobrze jest zrobić sobie funkcje pomocnicze i taki loger z prawdziwego zdarzenia, który rzeczywiście może pomóc w debugowaniu